feat: added typescript types to ClassPrefix component

SunnyJohal opened the pull request:
Closes #13547 

added typescript types to ClassPrefix component

#### Changelog

**New**

- added typescript types to ClassPrefix component

**Changed**

- file extension from .js to .tsx

**Removed**

- n/a

#### Testing / Reviewing

- Run storybook and check that the components are rendering.
- Run unit tests.

SunnyJohal committed:
feat: added typescript types to ClassPrefix component

mrkjdy committed:
feat: convert HeaderMenuButton.js to HeaderMenuButton.tsx and add types (#13739)

mrkjdy committed:
feat: convert HeaderNavigation.js to HeaderNavigation.tsx and add types (#13762)

tay1orjones commented on packages/react/src/components/ClassPrefix/index.tsx:
```diff
@@ -6,10 +6,21 @@
-function ClassPrefix({ children, prefix }) {
+interface ClassPrefixProps {
+  /**
+   * @description The value used to prefix the CSS selectors
```
The `@description` JSDoc annotation is [not supported](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#unsupported-tags)

```suggestion
   * The value used to prefix the CSS selectors
```

Additional usages of `@description` in this PR will need updated as well

tay1orjones reviewed (changes requested):
Sorry I just noticed this

renovate[bot] committed:
chore(deps): update yarn to v3.5.1 (#13756)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

mrkjdy committed:
feature/HeaderSideNavItems types (#13763)

* feat: convert HeaderSideNavItems.js to tsx and add types

* fix: update tests after removing default value from HeaderSideNavItems

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

lewandom committed:
refactor(types/common): change common types extension to .ts (#13751)

mrkjdy committed:
feature/SideNav types (#13765)

* feat: convert UIShell/SideNav.js to UIShell/SideNav.tsx and add types

* fix: fix formatting in SideNav.tsx

* fix: improve readability of adding customClassName in SideNav

* Update packages/react/src/components/UIShell/SideNav.tsx

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>

tay1orjones committed:
fix(vite): update example to use react 18 createRoot (#13720)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency rimraf to v5 (#13757)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

guidari committed:
docs(migration): prefix removal carbon-- v11 (#13738)

* docs(migration): prefix removal carbon--

* docs(migration): fix spacing

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

mrkjdy committed:
feature/Heading types (#13708)

* feat: convert Heading/index.js to Heading/index.tsx and add types

* fix: rename Heading/Heading-story.js to Heading/Heading.stories.js

* fix: replace Preview/Canvas with Props/ArgsTable in Heading/Heading.mdx

* fix: improve Section and Heading types

* fix: use section for default props on Section component

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

alisonjoseph committed:
docs: add more info around automated codemods to v11 migration (#13771)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

mrkjdy committed:
feature/Theme types (#13707)

* feat: convert Theme/index.js to Theme/index.tsx and add types

* fix: improve type for as prop in ThemeProps

* fix: improve as prop type in Theme

* fix: format Theme/index.tsx

* chore: add mrkjdy as contributor

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

alisonjoseph committed:
fix(search): scope contained list search styles (#13737)

* fix(search): scope contained list search styles

* chore: fix focus styles for contained list search

carbon-automation[bot] committed:
chore(release): v11.29.0-rc.0 (#13770)

Co-authored-by: tay1orjones <tay1orjones@users.noreply.github.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>

guidari committed:
fix(HeaderContainer): ssr fix by using useWindowEvent (#13781)

alisonjoseph committed:
feat(Slider): update invalid state add warning state (#13674)

* feat(Slider): update invalid state add warning state

* feat(slider): hide invalid and warn when readonly

* chore: update tests

* chore: cleanup storybook

* chore: fix lint

* Update packages/react/src/components/Slider/Slider.tsx

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>

* fix: invalid and readonly state

* Update packages/styles/scss/components/slider/_slider.scss

* feat: align text left update input width

---------

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

carbon-automation[bot] committed:
chore(release): v11.29.0 (#13787)

Co-authored-by: tay1orjones <tay1orjones@users.noreply.github.com>

mrkjdy committed:
feature/HeaderContainer types (#13717)

* feat: convert HeaderContainer.js to HeaderContainer.tsx and add types

* fix: use isActive instead of isCurrentPage in UIShell.stories.js

* fix: update PublicAPI-test.js.snap

* fix: format UIShell/HeaderContainer.tsx

* fix: improve UIShell/HeaderContainer render prop type

* fix: add doc comment explaining render prop in HeaderContainer

* fix: use correct proptype for the render prop of HeaderContainer

* fix: remove example from  doc comment on render prop of HeaderContainer

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>

guidari committed:
13420 rail sidenav esc (#13639)

* fix(SideNav): added esc fun

* fix(SideNav): reopen when press tab

* chore(SideNav): remove unused id

* fix(SideNav): switch the focus to main content

* fix(SideNav): added href prop to skip to main content

* fix(SideNav): fix tests

* fix(SideNav): added href as default prop

* fix(SideNav): fix snapshot test

* fix(SideNav): removed isRequired from href

* test(SideNav): fix snapshot test

* fix(SideNav): removed the default prop from href

* test(SideNav): fix snapshot test

* fix(SideNav): added an if to validate

---------

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>

falsepopsky committed:
chore(deps): bump and fix prettier (#13675)

* chore(deps): bump prettier to latest

* fix: format files with prettier

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

sjbeatle committed:
[a11y] Search: Search Icon Label Logic Update (#13648)

* [a11y] Search: Search Icon Label Logic Update

With the currently implementation, an accessibility violation occurs when the Search Icon's container doesn't have a role.
The current implementation removes the `role` attribute when the `onExpand` property is undefined, but it should also remove the `aria-labelledby`.

This update will avoid the accessibility violation: https://www.w3.org/TR/wai-aria-1.1/#roles

---

The error can be see in the `@carbon/react` Storybook UI. Visit the [Search component](https://react.carbondesignsystem.com/?path=/story/components-search--default) and click on the 'Accessibility' tab

* fix(tests): snapshots updated after Search a11y update

---------

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>
Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

kaiodealmeida committed:
fix(DatePicker): onChange callback not being called (#13774)

* fix: onChange callback not being called

add undefined as useSavedCallback param type and remove arrow func param

* fix(DatePicker): fix code format with prettier

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

tay1orjones committed:
Create release-schedule.md

tw15egan committed:
fix(Type): add v10 type tokens (#13776)

* fix(Type): add v10 type tokens to v11

* test(Type): add style test to ensure v10 type tokens are exported

* fix(Type): use correct namespace

* test(snapshot): update snapshots

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

tay1orjones committed:
fix(tabs): overflow scroll buttons should not be focusable (#13792)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

conradennis committed:
feat(icons): new icons february thru may (#13785)

* feat(icons): new icons february thru may

Submitting new and updated UI icons from February through May 2023
all updates and additions approved by BXD

30 net new icons

updated categories.yml with additions
updated ui-icon-master.ai with additions

* test(snapshot): update snapshots

* test(snapshot): update icons-vue snapshots

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

alisonjoseph committed:
docs: fix typo Update v11.md (#13779)

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Guilherme Datilio Ribeiro <guilhermedatilio@gmail.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

mrkjdy committed:
feature/Form types (#13773)

* feat: convert Form to tsx and update tests

* fix: export Form explicitly from Form/index.ts

* fix: update tests after exporting Form explicitly

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Taylor Jones <tay1orjones@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

mrkjdy committed:
feature/OrderedList, UnorderedList, and ListItem types (#13775)

* feat: convert OrderedList to tsx and update tests

* fix: formate OrderedList.tsx

* feat: convert UnorderedList to tsx and update tests

* fix: format UnorderedList.tsx

* feat: convert ListItem to tsx and update tests

* fix: export OrderedList/UnorderedList/ListItem explicitly, update tests

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

lewandom committed:
fix(TextInput): restrict onChange event TypeScript type (#13801)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

lewandom committed:
chore(enable-v11-release): cleanup enable-v11-release feature flag usage (#13735)

* chore(enable-v11-release): cleanup enable-v11-release feature flag usage

* fix(ComboBox): restore excessively removed classnames

* chore(Tab): cleanup classic Tab implementation

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

tw15egan committed:
fix(Layout): add v10 layout tokens (#13783)

* fix(Layout): add v10 layout tokens

* test(Layout): add tests for spacing tokens

* fix(Themes): add layout tokens to theme map

* test(snapshot): update snapshots

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

guidari committed:
fix(UIShell): loss of focus on menu (new) (#13796)

* fix(UIShell): new branch to fix merge problems

* test(snapshot): fix snapshots

lewandom committed:
fix(DataTable): correct default values for optional component props (#13804)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

jsehull committed:
fix(UIShell): add bg color to selected states (NEW) (#13402)

* refactor(UIShell): finish CSS for selected states

* fix(state): change parent state if child selected

* fix(UIShell): refactor some code

* fix(UIShell): more refactoring

---------

Co-authored-by: TJ Egan <tw15egan@gmail.com>

tw15egan committed:
fix(Motion): add v10 Motion tokens (#13805)

* fix(Motion): add v10 motion tokens

* test(Motion): add v10 compat tests

---------

Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>

TannerS committed:
fix(react): updated textarea counter value changes on re-render (#13449)

* fix(react): updated textarea counter value changes on re-render

* fix: format

* Update packages/react/src/components/TextArea/__tests__/TextArea-test.js

* Update packages/react/src/components/TextArea/__tests__/TextArea-test.js

* Update packages/react/src/components/TextArea/__tests__/TextArea-test.js

* Update packages/react/src/components/TextArea/__tests__/TextArea-test.js

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>

* fix(TextArea): use textarea ref value instead of [value,defaultValue]

* fix: format

* fix(TextArea): add value to textCounter dependency array

* fix(TextArea): textCount optimizations

- use value,defaultValue in useEffect insead of ref
- delay textCount assignation

* fix: format

---------

Co-authored-by: Francine Lucca <francinelucca@hotmail.com>
Co-authored-by: Andrea N. Cardona <cardona.n.andrea@gmail.com>
Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

andreancardona committed:
fix: a11y-dropdown-typeahead-downshift (#13744)

* feat: contained list persistent search

* docs: update docs for contained list search

* fix: fix border

* feat: functionality working

* fix(tabs): hover style bugs

* fix: remove hover from disabled state

* feat: clean up code

* Update packages/react/src/components/ContainedList/ContainedList.mdx

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>

* fix: update docs

* fix: fix styling

Co-authored-by: Francine Lucca <francinelucca@users.noreply.github.com>

* fix: update docs

* fix: failing test

* fix: fix tests

* fix: draft fix for dropdown

* fix: run format:

---------

Co-authored-by: Alison Joseph <alison.joseph@us.ibm.com>
Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>
Co-authored-by: Francine Lucca <francinelucca@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

mrkjdy committed:
feature/Toggletip types (#13790)

* feat: convert Toggletip and useEvent to tsx/ts

* fix: make Popover polymorphic

* fix: use correct syntax for importing in a few JS files

* fix: update tests after modifying types in Toggletip and Popover

* fix: revert useEvent to js

* fix: make useEvent and useWindowEvent generic

---------

Co-authored-by: Francine Lucca <40550942+francinelucca@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

mrkjdy committed:
feature/SkipToContent types (#13772)

* feat: convert UIShell/SkipToContent to tsx and update tests

* fix: require SkipToContent children to be a string

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

tay1orjones committed:
chore(security): add SECURITY.md policy (#13812)

* chore(security): add SECURITY.md policy

* chore(security): add SECURITY.md policy

* docs(security): reference the release schedule

---------

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

tw15egan committed:
chore(ClassPrefix): remove @description
